Item 7205: Dataset designer story4 - create dataset from a file

labkey-ankurj opened the pull request:
#### Rationale
This work provides the server side support for dataset creation by importing a file.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/243

labkey-ankurj committed:
Item 7205: Modify existing StudyController.ImportAction to utilize participantId and sequence num field

labkey-ankurj committed:
pulling in alpha version

labkey-ankurj committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205

labkey-ankurj committed:
pulling in alpha version

labkey-ankurj committed:
merge branch 'develop' into fb_datasetDesigner_story4_7205

cnathe commented on core/package.json:
```diff
@@ -90,8 +90,8 @@
   "dependencies": {
-    "@labkey/api": "0.2.8",
-    "@labkey/components": "0.56.3",
+    "@labkey/api": "0.2.4",
```
We don't want to downgrade this @labkey/api (we need it here for the LABKEY var in LK server). Nick bump this to pick up a fix for some automated test failures. Instead, you could pull in 0.2.8 in your @labkey/components package.json to make them match.

labkey-ankurj replied:
no idea how this got changed, will change it back.

cnathe replied:
It probably got reverted during a merge conflict resolution (i.e. merging develop into your fb).

labkey-ankurj replied:
done

labkey-ankurj replied:
I've it in the ui-components package.json, will merge that later.

labkey-ankurj committed:
pulling in alpha version

cnathe reviewed (changes requested):
The code change in StudyController looks good, but one change to revert in the core/package.json

labkey-ankurj committed:
put back @labkey/api version

labkey-ankurj committed:
pulling in alpha version

cnathe committed:
Update study dataset importData action comment related to columnMapping
- and change columnMap population so that we can support the demographics dataset scenario where only ParticipantId might be provided

labkey-ankurj committed:
correct column map for different studies

labkey-ankurj committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205

labkey-ankurj committed:
alpha version for cr-feedback and other fixes

labkey-ankurj committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205

labkey-ankurj committed:
pulling in alpha version

labkey-ankurj committed:
adding study/package.json

cnathe committed:
Fix for time as additional key validation check

cnathe committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205

cnathe committed:
Display file import error message in confirm modal for DatasetDesigner.tsx

cnathe committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205

cnathe committed:
Update to the latest @labkey/components alpha package version

cnathe committed:
Study dataset import column mapping fix for "Sequence Num" -> "SequenceNum"

cnathe committed:
Update to the latest @labkey/components alpha package version

cnathe committed:
Update to the latest @labkey/components alpha package version

cnathe committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205

cnathe committed:
Update to the latest @labkey/components alpha package version

labkey-ankurj committed:
Item 7205: update to release version of ui-components

labkey-ankurj committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205

labkey-ankurj committed:
Merge branch 'develop' into fb_datasetDesigner_story4_7205
